File: flutter_app_signalr_chat/lib/models/message_model.dart
class MessageModel {
  String message;
  String dateTime;

  MessageModel({this.message, this.dateTime});

  MessageModel.fromJson(Map<String, dynamic> json) {
    message = json['message'];
    dateTime = json['dateTime'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['message'] = this.message;
    data['dateTime'] = this.dateTime;
    return data;
  }
}

class ChatMessageModel extends MessageModel {
  String from;

  ChatMessageModel({this.from, String message, String dateTime})
      : super(message: message, dateTime: dateTime);

   ChatMessageModel.fromJson(Map<String,dynamic> json) {
     from = json['from'];
     message = json['message'];
     dateTime = json['dateTime'];
   }
}

File: SignalR_Chat/SignalR_Chat/Models/MessageModel.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SignalR_Chat.Models
{
    public class ChatMessageModel
    {
        public string from;
        public string message;
        public string dateTime;

        public ChatMessageModel(string from,string message, string dateTime)
        {
            this.from = from;
            this.message = message;
            this.dateTime = dateTime;
        }
        public string toJson()
        {
            return JsonConvert.SerializeObject(this);
        }
        public ChatMessageModel fromJson(String json)
        {
            return JsonConvert.DeserializeObject<ChatMessageModel>(json);
        }
    }
}
